cpp/oneapi/dal/algo/knn/common.hpp

```diff
@@ -72,6 +76,35 @@ using v1::by_default;
 
 } // namespace method
 
+/// Represents result option flag
+/// Behaves like a regular :expr`enum`.
+class result_option_id : public result_option_id_base {
+public:
+    result_option_id() : result_option_id_base{} {}
+    result_option_id(const result_option_id_base& base) : result_option_id_base{ base } {}
+};
+
+namespace detail {
+
+ONEDAL_EXPORT result_option_id get_indices_id();
+ONEDAL_EXPORT result_option_id get_distances_id();
+ONEDAL_EXPORT result_option_id get_responses_id();
+
+template <typename Task>
+ONEDAL_EXPORT result_option_id get_default_result_options();
```

Comment:
Why this method is declared in the header file? I don't see any usage here. The only place where it's used is in cpp. So, suggest removing this function from hpp.

Reply:
It is used both in `common.cpp` and `infer_types.cpp`

Reply:
As we discussed, `infer_types.cpp` and `infer_result` should not use default options.

Reply:
Fixed